var/spack/repos/builtin/packages/py-tensorboard/package.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyTensorboard(Package):
+    """TensorBoard is a suite of web applications for
+    inspecting and understanding your TensorFlow runs and
+    graphs."""
+
+    homepage = "https://pypi.python.org/project/tensorboard"
+    url      = "https://github.com/tensorflow/tensorboard/archive/1.12.2.tar.gz"
+
+    version('2.2.0', sha256='d0dfbf0e4b3b5ebbc3fafa6d281d4b9aa5478eac6bac3330652ab6674278ab77')
+
+    depends_on('bazel@0.26.1:', type='build')
+    depends_on('py-setuptools@41.0.0:', type='build')
+    depends_on('py-absl-py@0.4:', type=('build', 'run'))
+    depends_on('py-markdown@2.6.8:', type=('build', 'run'))
+    depends_on('py-requests@2.21.0:3.0.0', type=('build', 'run'))
+    depends_on('py-futures@3.1.1:', type=('build', 'run'), when='^python@:2')
+    depends_on('py-grpcio@1.23.3:', type=('build', 'run'))
+    depends_on('py-google-auth@1.6.3:1.99.99', type=('build', 'run'))
+    depends_on('py-numpy@1.12.0:', type=('build', 'run'))
+    depends_on('py-protobuf@3.6.0:', type=('build', 'run'))
+    depends_on('py-six@1.10.0:', type=('build', 'run'))
+    depends_on('py-werkzeug@0.11.15:', type=('build', 'run'))
+    depends_on('py-wheel', type=('build', 'run'))
+    depends_on('py-wheel@0.26:', type=('build', 'run'), when='@0.6: ^python@3:')
+#   depends_on('py-google-auth-oauthlib@0.4.1:0.5', type=('build', 'run'))
```

Comment:
Where did you find some of these dependencies? All I'm seeing is:
```python
REQUIRED_PACKAGES = [                                                           
    # futures is a backport of the python 3.2+ concurrent.futures module        
    'futures >= 3.1.1; python_version < "3"',                                   
    'grpcio >= 1.6.3',                                                          
    'markdown >= 2.6.8',                                                        
    'numpy >= 1.12.0',                                                          
    'protobuf >= 3.4.0',                                                        
    'six >= 1.10.0',                                                            
    'werkzeug >= 0.11.10',                                                      
    # python3 specifically requires wheel 0.26                                  
    'wheel; python_version < "3"',                                              
    'wheel >= 0.26; python_version >= "3"',                                     
]
```

Reply:
I worked from tensorboard/pip_package/requirements.txt in the git repo:

```
# Non-vendored runtime dependencies of TensorBoard.

absl-py >= 0.4
# futures is a backport of the python 3.2+ concurrent.futures module
futures >= 3.1.1; python_version < "3"
grpcio >= 1.24.3
google-auth >= 1.6.3, < 2
google-auth-oauthlib >= 0.4.1, < 0.5
markdown >= 2.6.8
numpy >= 1.12.0
protobuf >= 3.6.0
requests >= 2.21.0, < 3
setuptools >= 41.0.0
six >= 1.10.0
tensorboard-plugin-wit >= 1.6.0
werkzeug >= 0.11.15
# python3 specifically requires wheel 0.26
wheel; python_version < "3"
wheel >= 0.26; python_version >= "3"

```

Reply:
Ah, first of all, never trust `requirements.txt`. Also, it sounds like the master branch has additional requirements that aren't present in this tarball.

Reply:
I'll prune these back.

Reply:
ah, wait, you're looking at the 1.12.2 tarball? Look at the 2.2.0 tarball instead. setup.py looks roughly like this, and those are the deps called out in that.

```
def get_required_packages():
    with open("requirements.txt") as f:
        return f.read().splitlines()

REQUIRED_PACKAGES = get_required_packages()

```

Reply:
Ah yes, can you update to the URL for 2.2.0? It doesn't affect downloads, but it affects URL parsing stats.